### PR TITLE
[NIFI-13093] use warn palette for all errors and introduce caution palette

### PR DESCRIPTION
**Pull request by scottyaslan**

### CODE REVIEW

**Comment by rfellows** [comment]
reviewing...

File: nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/pages/bulletins/ui/bulletin-board/bulletin-board-list/bulletin-board-list.component.ts [hunk 1]
```diff
@@ -148,7 +148,7 @@ export class BulletinBoardList implements AfterViewInit {
             case 'warning':
                 return 'bulletin-warn';
```
Comment (rfellows) [inline nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/pages/bulletins/ui/bulletin-board/bulletin-board-list/bulletin-board-list.component.ts:149]: should this one use the `caution-color`

**Review by rfellows — COMMENTED** [review]
just a few things.

File: nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/pages/bulletins/ui/bulletin-board/bulletin-board-list/bulletin-board-list.component.ts (hunk 1 above)
Comment (scottyaslan) [reply nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/pages/bulletins/ui/bulletin-board/bulletin-board-list/bulletin-board-list.component.ts:149]: No. Bulletins have a `.bulletin-error` state that is red. This is correct.

File: nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/assets/styles/_app.scss [hunk 2]
```diff
@@ -261,56 +281,43 @@
...
         color: $nifi-theme-success-palette-darker;
         fill: $nifi-theme-success-palette-darker;
     }
 
-    .nifi-warn-default {
-        color: $nifi-theme-warn-palette-default;
-        fill: $nifi-theme-warn-palette-default;
+    .caution-color {
+        color: $nifi-theme-caution-palette-default;
+        fill: $nifi-theme-caution-palette-default;
```
[the remark this answers is not in the document]
Comment (scottyaslan) [reply nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/assets/styles/_app.scss]: Pretty much any time we need a yellow color it should be the caution darker color. Currently this is used extensively for any `.fa .fa-warning` icons but it is also used for things like back pressure, when a connection is selected on the canvas, and midpoint/bendpoints/label resize.

File: nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/assets/themes/material.scss [hunk 3]
```diff
@@ -50,8 +50,33 @@ $material-primary-light-palette: (
...
+        200: rgba(#000000, 0.87),
+        300: rgba(#000000, 0.87),
+        400: #ffffff,
+        500: #ffffff,
+        600: #ffffff,
+        700: #ffffff,
+        800: #ffffff,
+        900: #ffffff
+    )
+);
```
[the remark this answers is not in the document]
Comment (scottyaslan) [reply nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/assets/themes/material.scss:79]: Good catch! I have updated this.

**Comment by scottyaslan** [comment]
> Overall this looks great. I did find one thing that we might want to clean up but I'm also happy if we just want to log it and fix as part of another pull request...
> 
> Essentially, in dark mode, the checkbox we use to support select all, select none, or indicate some are selected isn't clearly defined when checked or indicating some are selected. It isn't immediately apparent that you can interact with it. It is fine when nothing is selected because it has an outline:
> 
> Screenshot 2024-05-02 at 8 30 28 AM Screenshot 2024-05-02 at 8 31 21 AM Screenshot 2024-05-02 at 8 31 34 AM
> @scottyaslan let me know if you want to include a fix for this here or if we should log it separately.

Ooooooo good eye! Let's log it as a separate jira. Thanks!

**Review by rfellows — APPROVED** [review]

**Pull request merged**